docs/design/satellite-svc-separation.md

```diff
@@ -0,0 +1,208 @@
+# Satellite Service Separation
+
+## Table of Contents
+1. [Abstract](#abstract)
+2. [Background](#background): Current Satellite Design
+3. [Design](#design)
+4. [Rationale](#rationale)
+4. [Implementation](#implementation)
+5. [Open Issues](#open-issues)
+
+## Abstract
+
+The goal of the design doc is to describe the work necessary to make the Satellite horizontally scalable.
+
+## Background
+
+Currently all Satellite services run in a single binary in a single container. While this is great for development, this is bad for scaling.
+
+### Current Satellite design
+
+Currently the Satellite is a single binary made of up the following services:
+
+#### overlay
+The overlay cache maintains critical information about nodes in the network (via the nodes table in Satellite.DB). One important use of overlay is to help select which storage nodes to upload files to.
+
+#### metainfo
+Metainfo is responsible for all things related to the metainfo stored for each file on the network. The metainfo system is currently composed of the following parts: 
+
+1) metainfoDB, a key/value store with segment path names as the key and a pointer as the value, 
+
+2) metainfo loop which iterates over all the key/values in metainfoDB, 
+
+3) metainfo endpoint which creates the public RPCs for creating/deleting metainfo, 
+
+4) metainfo service.
```

Comment:
metainfo service, which ensures proper access to metainfo database and simplifies modification for other services.